src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,114 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    dst_f8->f1[0] = __logf(fabsf(src_f8->f1[0]));
+    dst_f8->f1[1] = __logf(fabsf(src_f8->f1[1]));
+    dst_f8->f1[2] = __logf(fabsf(src_f8->f1[2]));
+    dst_f8->f1[3] = __logf(fabsf(src_f8->f1[3]));
+    dst_f8->f1[4] = __logf(fabsf(src_f8->f1[4]));
+    dst_f8->f1[5] = __logf(fabsf(src_f8->f1[5]));
+    dst_f8->f1[6] = __logf(fabsf(src_f8->f1[6]));
+    dst_f8->f1[7] = __logf(fabsf(src_f8->f1[7]));
+
+    if constexpr (std::is_same<T, schar>::value)
+    {
+        dst_f8->f4[0] = rpp_hip_pixel_check_0to255(dst_f8->f4[0]) - (float4)128;
+        dst_f8->f4[1] = rpp_hip_pixel_check_0to255(dst_f8->f4[1]) - (float4)128;
+    }
+}
+
+// F16 stores without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store(half *dstPtr, d_float8 *dst_f8)
+{
+    d_half8 dst_h8;
+
+    dst_h8.h2[0] = __float22half2_rn(make_float2(dst_f8->f1[0], dst_f8->f1[1]));
+    dst_h8.h2[1] = __float22half2_rn(make_float2(dst_f8->f1[2], dst_f8->f1[3]));
+    dst_h8.h2[2] = __float22half2_rn(make_float2(dst_f8->f1[4], dst_f8->f1[5]));
+    dst_h8.h2[3] = __float22half2_rn(make_float2(dst_f8->f1[6], dst_f8->f1[7]));
+
+    *(d_half8 *)dstPtr = dst_h8;
```

Comment:
we already had this function in rpp_hip_common.hpp
https://github.com/ROCm/rpp/blob/master/src/include/hip/rpp_hip_common.hpp#L979

Why it has been readed here? Anything has been changed?

Reply:
As templated function is used below had to introduce one again for f16 similar to that found in hip_common

Reply:
The output datatype is implicit from template argument T2 which is for output
So for U8, I8, F32 user will pass float buffers for outputs
for f16, user will pass f16 buffers as outputs

So when you call  rpp_hip_pack_float8_and_store(dstPtr + dstIdx, &dst_f8); 
for U8, I8, F32 float overloaded function in rpp_hip_common.hpp will be called

For F16
f16 overloaded function in rpp_hip_common.hpp will be called. So again why this function is needed then?

Reply:
common function there is named as rpp_hip_pack_float8_and_store8, thats why i named mine as rpp_hip_pack_float8_and_store

Reply:
Let's have a call on this

Reply:
Removed extra pack functions